### DISCUSSION TITLE
Not working with TFVC

### DISCUSSION THREAD

**Discussion by DinTri** [rfennell/AzurePipelines#1297] [Q&A] [answered]
Hi Richard,
Our company doesn't use Git and Pull Requests. We use TFVC
Is it possible to generate release notes only from ChangeSets and Boards with Work Items, Bugs?
I tried the console application locally but I receive:
##vso[task.debug]task result: Failed
##vso[task.issue type=error;]Unhandled: Cannot read properties of null (reading 'value')
##vso[task.complete result=Failed;]Unhandled: Cannot read properties of null (reading 'value')

**Comment by rfennell** [comment, marked as answer]
Yes, it is possible, it is just that the Git related objects will not be populated. You end up with a template at it's simplest something like this

```
## Build {{buildDetails.buildNumber}}
* **Completed**: {{buildDetails.finishTime}}
* **Previous Build**: {{compareBuildDetails.buildNumber}}

# Global list of WI with PRs, parents and children
{{#forEach this.workItems}}
{{#if isFirst}}### WorkItems {{/if}}
*  **{{this.id}}**  {{lookup this.fields 'System.Title'}}
   - **WIT** {{lookup this.fields 'System.WorkItemType'}}
   - **Tags** {{lookup this.fields 'System.Tags'}}
   - **Assigned** {{#with (lookup this.fields 'System.AssignedTo')}} {{displayName}} {{/with}}
   - **Description** {{{lookup this.fields 'System.Description'}}}
{{/forEach}}

# Global list of CS ({{commits.length}})
{{#forEach commits}}
{{#if isFirst}}### Associated commits{{/if}}
* ** ID{{this.id}}**
   -  **Message:** {{this.message}}
{{/forEach}}
```

If you set a build variable `system.debug=true` you can see all the processes used to get the details and what objects are populated.

As to the local tester, look like you have a misconfigured config file.

**Reply by DinTri** [reply]
Thank you Richard.
Only this part is not working:
# Global list of WI with PRs, parents and children
Here is what I receive:
Associated work items
None
Is there any way to make it works?
